universe/Condition.cpp

```diff
@@ -10092,8 +10092,11 @@ void OrderedAlternativesOf::Eval(const ScriptingContext& parent_context,
             FCMoveContent(temp_objects, matches);
         }
 
-        // No operand condition was selected. Nothing needed to be applied. State is restored. Done.
+        // No operand condition was selected. State is restored...
     }
+    // ...  Nothing should match, so move all to non_matches input set.
+    non_matches.reserve(matches.size() + non_matches.size());
+    FCMoveContent(matches, non_matches);
```

Comment:
Does it make sense to always do the samething here, independent if thesearch domain? If searching `NON_MATCHES`, the contents of `matches` shouldn't be modified.

Reply:
Hm. Probably you are right. For NON-MATCHES it should be a no-op. Not sure what a test case would be

Reply:
Hm. Probably you are right. For NON-MATCHES it should be a no-op. Not sure what a test case would be

Reply:
Found a test case for NON_MATCHES domain
`Or [ Building OrderedAlternativesOf [ Ship ] ]`

and the no-op fixes the bug